pkg/controller/operands/networkAddons_test.go

```diff
@@ -362,6 +362,81 @@ var _ = Describe("CNA Operand", func() {
 			Expect(req.Conditions).To(BeEmpty())
 		})
 
+		It("should have ovs if deployOVS annotation is set to true", func() {
+			hcoOVSConfig := commonTestUtils.NewHco()
+			hcoOVSConfig.Annotations = map[string]string{}
+			hcoOVSConfig.Annotations["deployOVS"] = "true"
+
+			existingResource := NewNetworkAddons(hcoOVSConfig)
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Updated).To(BeTrue())
+			Expect(res.Err).To(BeNil())
+
+			foundResource := &networkaddonsv1.NetworkAddonsConfig{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			Expect(existingResource.Spec.Ovs).ToNot(BeNil())
+			Expect(req.Conditions).To(BeEmpty())
+
+		})
+
+		It("should not have ovs if deployOVS annotation is not set to true", func() {
+			hcoOVSConfig := commonTestUtils.NewHco()
+			hcoOVSConfig.Annotations = map[string]string{}
+			hcoOVSConfig.Annotations["deployOVS"] = "false"
+
+			existingResource := NewNetworkAddons(hcoOVSConfig)
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+			Expect(res.UpgradeDone).To(BeFalse())
```

Comment:
why does the first test have `Expect(res.Updated).To(BeTrue())` and the other 2 don't? is it intentional?

Reply:
Because the first test case is updating the CR (adding the OVS field), while the two other are not changing it (OVS is not in the CNAO CR).